File: 20200108/skilltree.test.js
const solution = (skill, skill_trees) =>
  skill_trees
    .map(v => v.replace(new RegExp(`[^${skill}]`, 'g'), ''))
    .filter(i => skill.startsWith(i))
    .length;

test('solution', () => {
  expect(solution('CBD', ['CBADF'])).toBe(1)
  expect(solution('CBD', ['BACDE', 'CBADF', 'AECB', 'BDA'])).toBe(2)
  expect(solution('ABC', ['EFG'])).toBe(1)
});

File: 20200315/python/skilltree_test.py
import re


def solution(skill, skill_trees):
    skills = [re.sub('[^{}]'.format(skill), '', s) for s in skill_trees]
    return len([s for s in skills if skill.startswith(s)])


def test_solution():
    assert solution('CBD', ['BACDE', 'CBADF', 'AECB', 'BDA']) == 2
